src/test/java/org/mongojack/TestDBUpdateSerialization.java

```diff
@@ -182,4 +206,71 @@ public void serialize(String value, JsonGenerator jgen,
             }
         }
     }
+    
+    public static class NestedRepeatedAttributeName {
+    	public static class Inner {
+    		public Date timestamp;
+
+			@Override
+			public int hashCode() {
+				final int prime = 31;
+				int result = 1;
+				result = prime * result
+						+ ((timestamp == null) ? 0 : timestamp.hashCode());
+				return result;
+			}
+
+			@Override
+			public boolean equals(Object obj) {
+				if (this == obj)
+					return true;
+				if (obj == null)
+					return false;
+				if (getClass() != obj.getClass())
+					return false;
+				Inner other = (Inner) obj;
+				if (timestamp == null) {
+					if (other.timestamp != null)
+						return false;
+				} else if (!timestamp.equals(other.timestamp))
+					return false;
+				return true;
+			}
+    	}
+    	public String _id = "id";
+    	public Inner inner = new Inner();
+    	public long  timestamp;
+		@Override
+		public int hashCode() {
+			final int prime = 31;
+			int result = 1;
+			result = prime * result + ((_id == null) ? 0 : _id.hashCode());
```

Comment:
no need for parens in `(_id == null)`.  same with the ternary operator on the next line